### ISSUE TITLE
Alt/caption for image field types 

### ISSUE THREAD

**Issue by GwendolenLynch** [bolt/bolt#827]
Image and imagelist field types should have an additional text field to allow for linked caption/alt text to be associated with the image.  

The alt="" has SEO importance and it would be nice to be able to have a caption frame automatically generated for the image that can be styled by the selected/used user theme.

Something like this would be cooler than ice:

```
fields:
        image:
            type: image
            alt: true
            caption: true
```

The booleans will give template designers to option to use or ignore.

**Comment by gpkvt** [comment]
The first comment says "Image and imagelist". Am I correct that this feature is available for image-fields only, yet?

At least I can't find a way to enter a title for an imagelist-item (using bolt 2.2.7). Would be nice if this could get added.

**Comment by GawainLynch** [comment]
In `imagelist`s, after you've added them to the list, you can just click on the file name and change that text. I just defaults to the file name.
![imagelist](https://cloud.githubusercontent.com/assets/1427081/9276283/c9145b30-429a-11e5-8995-28e10f0af7eb.png)

**Comment by gpkvt** [comment]
Well that's embarrassing. ;) Thanks!

**Comment by GawainLynch** [comment]
Nah, it's just not obvious… and that popover text in the **i** could be better

**Comment by edanao** [comment]
is there by any chance that we could add like a description and a link to the images in the imagelist? :)

**Comment by SahAssar** [comment]
@edanao If you want more fields in the imagelist you might want to look into making it a repeating field with an image field and the additional fields that you want. So in your repeater you'd have a imagefield, a textfield, a textarea and a field for your link, then it should work pretty much how you want it to.

**Comment by edanao** [comment]
thank you @SahAssar !!!

**Comment by MikSDigital** [comment]
Hello!

By the way, is it possible to call alt attribute with this statement:
`{{ showimage(post.image, 500,360) }}`
?

**Comment by SahAssar** [comment]
@MikSDigital This issue is 3 years old and that sort of question is probably better for our slack/irc or the discusssion boards, but to answer your question that should work, yes.

**Comment by nestordedios** [comment]
Hola everyone!! 

I know this is an old closed issue but I did not want to open a new issue for something that might be related. 

When I use this configuration:
```
pages:
    ...
    fields:
        ...
        image:
            type: image
            attrib: [title, alt]
        slider:
            type: imagelist
            attrib: [title, alt]
        ...
```
Bolt 3.3.6 backend looks like this:
![image imagelist](https://user-images.githubusercontent.com/6607455/32599056-a7bbe50e-c53b-11e7-9334-e57ba1b337a5.png)

For imagelist items I don't have the option to edit `alt` and `title` attributes. I can only edit text to be used either as `alt` or `title`. In the other hand, for a single image field I can edit both attributes. I'm not the one to say, but maybe this issue needs to be reopened?  

I have this issue on Bolt 2.2.22 and Bolt 3.3.6

**Comment by GawainLynch** [comment]
Hey mate. Nope, this is a limitation of `imagelist`s. It would be good to review how we handle fields types more consistently in v4, but for v3 it would break BC.

**Comment by nestordedios** [comment]
Hey @GawainLynch That's a pitty :(. Then I think using repeater fields is the best way to go for this. Looking forward for those improvements when v4 lands :). Thanks 👍
